utils/evaluator.py

```diff
@@ -34,17 +39,40 @@ def __call__(self, data_loader):
             self.current_intersection = torch.sum(prediction * ground_truth)
             self.total_pixels = torch.sum(ground_truth) + torch.sum(prediction)
             self.current_union = self.total_pixels - self.current_intersection
-            total_iou += self.iou()
-            total_f_score += self.f_score()
-        return total_iou / len(data_loader), total_f_score / len(data_loader)
+            total_iou += self.__iou()
+            total_f_score += self.__f_score()
+        self.mean_iou = total_iou / len(data_loader)
+        self.mean_f_score = total_f_score / len(data_loader)
+        self.dataset_name = type(data_loader.dataset).__name__
+        self.add_row_to_table()
 
-    def iou(self):
+    def __iou(self):
         return self.current_intersection / \
                ((self.current_union - self.current_intersection) + self.epsilon)
 
-    def f_score(self):
+    def __f_score(self):
         return 2 * self.current_intersection / (self.total_pixels + self.epsilon)
```

Comment:
You already made the static functions that handle them, why is the rewriting ?

Reply:
to avoid computing the intersection and union twice thus taking twice the time to compute the metrics